### PR TITLE
Refactor code for official release

### PR DESCRIPTION
**Pull request by XinhuiLi**
- [x] Replace local paths with variables with descriptive names
- [x] Remove comments in the code

### CODE REVIEW

File: figure/3/extract_ROI.sh [hunk 1]
```diff
@@ -57,18 +44,17 @@ for run in 'cpac_abcd_all' 'cpac_fmriprep_all';do
...
 rm $commandlist
 
 for num_rois in 200;do
     echo $num_rois
-    mni2004_2mm='/data3/cnl/fmriprep/Lei_working/Finalizing/Schaefer_Atlas/Schaefer2018_'$num_rois'Parcels_7Networks_order_FSLMNI152_2mm.nii.gz'
-    mni2004_3mm='/data3/cnl/fmriprep/Lei_working/Finalizing/Schaefer_Atlas/Schaefer2018_'$num_rois'Parcels_7Networks_order_FSLMNI152_3mm.nii.gz'
-    mni2009='/data3/cnl/fmriprep/Lei_working/Finalizing/Schaefer_Atlas/Schaefer2018_'$num_rois'Parcels_7Networks_order_FSLMNI152_2mm_NLin2009cAsym.nii.gz'
+    mni2004_2mm=$WORKING_DIR'/Schaefer_Atlas/Schaefer2018_'$num_rois'Parcels_7Networks_order_FSLMNI152_2mm.nii.gz'
+    mni2004_3mm=$WORKING_DIR'/Schaefer_Atlas/Schaefer2018_'$num_rois'Parcels_7Networks_order_FSLMNI152_3mm.nii.gz'
+    mni2009=$WORKING_DIR'/Schaefer_Atlas/Schaefer2018_'$num_rois'Parcels_7Networks_order_FSLMNI152_2mm_NLin2009cAsym.nii.gz'
```
Comment (shnizzedy) [inline figure/3/extract_ROI.sh]: Are `Shaefer_ROI_commands.txt` and these atlases included or linked somewhere?

Comment (shnizzedy) [reply figure/3/extract_ROI.sh]: Also `Shaefer` or `Schaefer`?

File: figure/3/plot_fig3_gsr.py [hunk 2]
```diff
@@ -8,9 +6,9 @@
 import seaborn as sns
 import matplotlib.pyplot as plt
 
-axis_tick_size=17
-lw_value=2
-to_sum = False#True
+agg_icc=True # whether aggregate ICC output or not, set it to 'True' for the first time and 'False' otherwise
+datain='ICC_1000_All_pearson' # path to ICC output from run_ICC.py
```
Comment (shnizzedy) [inline figure/3/plot_fig3_gsr.py]: I'd use an environment variable here too, like

```Python
datain = os.environ.get('RUN_ICC_OUTPUT')
```

File: figure/1/Reliability_general_minimal.R [hunk 3]
```diff
@@ -290,7 +290,7 @@ iccplot <- function(data,outfile){
   
 }
 
-basefolder='/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis'
+basefolder='/root'
```
Comment (shnizzedy) [inline figure/1/Reliability_general_minimal.R]: I'd use an environment variable here, like 

```R
basefolder <- Sys.getenv("BASEFOLDER")
```

File: figure/1/extract_ROI.sh [hunk 4]
```diff
@@ -1,6 +1,12 @@
+# TODO define variables
+DATA_INPUT_DIR=''
+DATA_OUTPUT_DIR=''
+WORKING_DIR=''
```
Comment (shnizzedy) [inline figure/1/extract_ROI.sh]: I'd define these outside of this script, although I think it's a good idea to include a comment at the top of each script and/or a usage string to say which environment variables that script uses.

File: figure/1/plot_fig1.py [hunk 5]
```diff
@@ -4,15 +4,15 @@
 fc_handle=''
 simpleplot=False
 atlases=['200','600','1000']
-base='/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis/ROI'
+base='/root/ROI'
```
Comment (shnizzedy) [inline figure/1/plot_fig1.py]: I'd use an environment variable here, like 

```Python
base = f'{os.environ.get("BASEFOLDER")}/ROI'
```

File: figure/1/run_ICC.sh [hunk 6]
```diff
@@ -1,3 +1,5 @@
+WORKING_DIR=''
```
Comment (shnizzedy) [inline figure/1/run_ICC.sh]: I'd define this outside of this script, although I think it's a good idea to include a comment at the top of each script and/or a usage string to say which environment variables that script uses.

File: figure/2/plot_fig2.py [hunk 7]
```diff
@@ -452,19 +452,19 @@ def label(x, color, label):
     simpleplot=False
     atlases=['200','600','1000']
 
-    wd_path='/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis'
+    wd_path='/root'
```
Comment (shnizzedy) [inline figure/2/plot_fig2.py]: I'd use an environment variable here, like 

```Python
wd_path = f'{os.environ.get("BASEFOLDER")}'
```

File: figure/3/extract_ROI.sh [hunk 8]
```diff
@@ -1,5 +1,10 @@
-datain='/data3/cnl/xli/reproducibility/out/gsr'
-dataout='/data3/cnl/fmriprep/Lei_working/Finalizing/All_sessions_new/data'
+DATA_INPUT_DIR=''
+DATA_OUTPUT_DIR=''
+WORKING_DIR=''
```
Comment (shnizzedy) [inline figure/3/extract_ROI.sh]: I'd define these outside of this script, although I think it's a good idea to include a comment at the top of each script and/or a usage string to say which environment variables that script uses.

File: figure/3/plot_fig3_time.py [hunk 9]
```diff
@@ -8,9 +6,9 @@
 import seaborn as sns
 import matplotlib.pyplot as plt
 
-axis_tick_size=17
-lw_value=2
-to_sum = False#True
+agg_icc=True # whether aggregate ICC output or not, set it to 'True' for the first time and 'False' otherwise
+datain='ICC_1000_All_pearson' # path to ICC output from run_ICC.py
```
Comment (shnizzedy) [inline figure/3/plot_fig3_time.py]: I'd use an environment variable here too, like

```Python
datain = os.environ.get('RUN_ICC_OUTPUT')
```

File: figure/3/run_ICC.py [hunk 10]
```diff
@@ -640,7 +640,7 @@ def list_difference(list1,list2):
 atlases=['200',]
 bootstrapping_ci=False
 bootstrapping_average=False
-base='/data3/cnl/fmriprep/Lei_working/testing/ICC_Scan_duration/All_sessions/ROI'
+base='/ROI'
```
Comment (shnizzedy) [inline figure/3/run_ICC.py]: I'd use an environment variable here like

```Python
base = f'{os.environ.get("ROI_BASE")}/ROI'
```

File: figure/3/run_ICC_I2C2.sh [hunk 11]
```diff
@@ -1,32 +1,29 @@
-## creat data for ICC and I2C2
+WORKING_DIR=''
```
Comment (shnizzedy) [inline figure/3/run_ICC_I2C2.sh]: I'd define this outside of this script, although I think it's a good idea to include a comment at the top of each script and/or a usage string to say which environment variables that script uses.

File: figure/4/corr_harm.py [hunk 12]
```diff
@@ -28,8 +28,8 @@ def cal_corr(path1, path2, run):
         sc.append(np.corrcoef(corr1_new,corr2_new)[0,1])
     np.save(os.path.join(out_path, run+'.npy'), sc)
 
-root_path = '/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis/ROI/ROI_Schaefer200/'
-out_path = '/Users/xinhui.li/Documents/reproducibility/reproducibility/figure/4/PearsonCorr'
+root_path = '/root/ROI/ROI_Schaefer200/'
+out_path = 'figure/4/PearsonCorr'
```
Comment (shnizzedy) [inline figure/4/corr_harm.py]: I'd use environment variables here, like 

```Python
root_path = f'{os.environ.get("BASEFOLDER")}/ROI/ROI_Schaefer200'
out_path = f'{os.environ.get("LOCAL_ROOT")}/figure/4/PearsonCorr'
```

File: figure/4/plot_fig4.ipynb [hunk 13]
```diff
@@ -32,7 +32,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "path = '/Users/xinhui.li/Documents/reproducibility/reproducibility/figure/4/PearsonCorr'\n",
+    "path = 'figure/4/PearsonCorr'\n",
```
Comment (shnizzedy) [inline figure/4/plot_fig4.ipynb]: I'd use an environment variables here, like 

```Python
path = f'{os.environ.get("LOCAL_ROOT")}/figure/4/PearsonCorr'
```

File: figure/5/plot_fig5.py [hunk 14]
```diff
@@ -5,7 +5,7 @@
 simpleplot=False
 atlases=['200','600','1000']
 
-wd_path='/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis'
+wd_path='/root'
```
Comment (shnizzedy) [inline figure/5/plot_fig5.py]: I'd use an environment variable here, like 

```Python
wd_path = os.environ.get("BASEFOLDER")
```

File: figure/s1/abcd.py [hunk 15]
```diff
@@ -139,8 +139,4 @@ def get_motion_params(file, pipeline = 'cpac'):
             corrs[num_sub][num_var+5] = round(corr, 3)
 
 print(corrs)
-np.save('/data3/cnl/xli/reproducibility/script/abcd_corrs.npy', corrs)
-
-# python /data3/cnl/CPAC_regtest_pack/corr_two_ts.py \
-# /data3/cnl/fmriprep/Lei_working/CPAC_XCP/ABCD/preprocessed/data/sub-0025427/ses-1/files/MNINonLinear/Results/task-rest01/task-rest01.nii.gz \
-# /data3/cnl/freesurfer/abcd-hcp_04-16-21/output/cpac_cpac_abcd-options/sub-0025427_ses-1/func/sub-0025427_ses-1_task-rest_run-1_space-template_desc-brain_bold.nii.gz
\ No newline at end of file
+np.save('/data3/cnl/xli/reproducibility/script/abcd_corrs.npy', corrs)
```
Comment (shnizzedy) [inline figure/s1/abcd.py]: I'd use an environment variable here, like 

```Python
np.save(f'{os.environ.get("SCRIPT_DIR")}/abcd_corrs.npy', corrs)
```

File: figure/s1/ccs.py [hunk 16]
```diff
@@ -143,28 +163,4 @@ def get_motion_params(file, pipeline = 'cpac'):
...
-
-# func2std (minimal) func1
-# applywarp --ref=/data3/cnl/xli/cpac_features/ccs/code/Ting/CCS/templates/MNI152_T1_3mm.nii.gz --in=rest_gms.nii.gz --out=rest_gms_mni152.nii.gz --warp=../anat/reg/highres2standard_warp.nii.gz --premat=reg/example_func2highres.mat
-
-# ccs_02_xt_funcbbregister.sh
-# convert_xfm -omat flirt.mat -concat flirt_rsp2rsp.mtx rpi2rsp.mat
-# cp flirt.mat example_func2highres.mat
-# convert_xfm -inverse -omat highres2example_func.mat example_func2highres.mat
\ No newline at end of file
+np.save('/data3/cnl/xli/reproducibility/script/ccs_corrs.npy', corrs)
```
Comment (shnizzedy) [inline figure/s1/ccs.py]: I'd use an environment variable here, like 

```Python
np.save(f'{os.environ.get("SCRIPT_DIR")}/ccs_corrs.npy', corrs)
```

File: figure/s1/plot_figs1.ipynb [hunk 17]
```diff
@@ -173,7 +173,7 @@
     "  for j in ses:\n",
     "      abcd_sub_list.append('sub-00'+str(i)+'_ses-'+str(j))\n",
     "\n",
-    "abcd_corrs = np.load('/Users/xinhui.li/Documents/reproducibility/reproducibility/figure/s1/abcd_corrs_100321.npy').T\n",
+    "abcd_corrs = np.load('figure/s1/abcd_corrs_100321.npy').T\n",
```
Comment (shnizzedy) [inline figure/s1/plot_figs1.ipynb]: I'd use an environment variable here, like 

```Python
abcd_corrs = np.load(f'{os.environ.get("LOCAL_ROOT")}/reproducibility/figure/s1/abcd_corrs_100321.npy').T
```

File: figure/s1/plot_figs1.ipynb [hunk 18]
```diff
@@ -201,7 +201,7 @@
     "  for j in ses:\n",
     "      ccs_sub_list.append('sub-00'+str(i)+'_ses-'+str(j))\n",
     "\n",
-    "ccs_corrs = np.load('/Users/xinhui.li/Documents/reproducibility/reproducibility/figure/s1/ccs_corrs_100321.npy').T\n",
+    "ccs_corrs = np.load('figure/s1/ccs_corrs_100321.npy').T\n",
```
Comment (shnizzedy) [inline figure/s1/plot_figs1.ipynb]: I'd use an environment variable here, like 

```Python
ccs_corrs = np.load(f'{os.environ.get("LOCAL_ROOT")}/reproducibility/figure/s1/ccs_corrs_100321.npy').T
```

File: figure/s1/plot_figs1.ipynb [hunk 19]
```diff
@@ -227,7 +227,7 @@
     "  for j in ses:\n",
     "      fmriprep_sub_list.append('sub-00'+str(i)+'_ses-'+str(j))\n",
     "\n",
-    "fmriprep_corrs = np.load('/Users/xinhui.li/Documents/reproducibility/reproducibility/figure/s1/fmriprep_corrs_100721.npy').T\n",
+    "fmriprep_corrs = np.load('figure/s1/fmriprep_corrs_100721.npy').T\n",
```
Comment (shnizzedy) [inline figure/s1/plot_figs1.ipynb]: I'd use an environment variable here, like 

```Python
fmriprep_corrs = np.load(f'{os.environ.get("LOCAL_ROOT")}/reproducibility/figure/s1/fmriprep_corrs_100721.npy').T
```

File: figure/s2/corr_fd.py [hunk 20]
```diff
@@ -8,9 +8,6 @@
     lines = f.readlines()
     for line in lines:
         sub_list.append(line.strip('\n').replace('_', '-'))
-sub_list.remove('sub-000000')
-sub_list.remove('sub-5463599') # crash
-sub_list.remove('sub-5713772') # crash
 
 good_sub_list = []
 good_sub_list_file = '/data3/cnl/fmriprep/Lei_working/FD_testing/HBN_testing/good_30.txt'
```
Comment (shnizzedy) [inline figure/s2/corr_fd.py]: I'd use an environment variable here, like 

```Python
good_sub_list_file = os.environ.get("GOOD_SUB_LIST_FILE")
```

File: figure/s2/plot_figs2.ipynb [hunk 21]
```diff
@@ -198,7 +198,7 @@
     "    for j in range(4-i,4):\n",
     "        axs[i, j].axis('off')\n",
     "\n",
-    "path = '/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis/ROI/ROI_Schaefer200'\n",
+    "path = '/root/ROI/ROI_Schaefer200'\n",
```
Comment (shnizzedy) [inline figure/s2/plot_figs2.ipynb]: I'd use an environment variable here, like 

```Python
path = f'{os.environ.get("BASEFOLDER")}/ROI/ROI_Schaefer200'
```

File: figure/s3/plot_figs3.py [hunk 22]
```diff
@@ -5,15 +5,15 @@
 simpleplot=False
 atlases=['200','600','1000']
 
-base='/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis/ROI'
+base='/root/ROI'
```
Comment (shnizzedy) [inline figure/s3/plot_figs3.py]: I'd use an environment variable here, like 

```Python
base = f'{os.environ.get("BASEFOLDER")}/ROI'
```

File: figure/s4/plot_figs4.py [hunk 23]
```diff
@@ -464,7 +464,7 @@ def label(x, color, label):
     simpleplot=False
     atlases=['200','600','1000']
 
-    wd_path='/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis'
+    wd_path='/root'
```
Comment (shnizzedy) [inline figure/s4/plot_figs4.py]: I'd use an environment variable here, like 

```Python
wd_path = os.environ.get("BASEFOLDER")
```

File: figure/utils/utils.py [hunk 24]
```diff
@@ -470,7 +470,7 @@ def label(x, color, label):
     ridgeplot(df,atlases,outpath + '/Ridgeplot_ICC_'+'_'+'-'.join(pipelines)[0:90]+'_'+atlas+'.png',plotnameorder)
 
 
-def cute_ani():
-    '''This is a test function written by cute Ani
+def func():
+    '''This is a test function written by dear Anibal Sólon
     '''
     print('Hello Pikachu!')
```
Comment (shnizzedy) [inline figure/utils/utils.py]: Hello Pikachu!

**Review by shnizzedy — COMMENTED** [review]
I'd create a [run commands file](https://en.wikipedia.org/wiki/Run_commands) like

```BASH
#!/bin/bash
# Set these variables and run `source .pipelineharmonizationrc` prior to running PipelineHarmonization scripts
export PH_SERVER_ROOT="/data3/cnl"  # folder on server containing relevant files
export LOCAL_ROOT="/Users/xinhui.li/Documents/reproducibility"  # folder on local machine containing relevant files
export BASEFOLDER="${LOCAL_ROOT}/LA/Reproducibility_Analysis"
export LOCAL_OUTPUT_DIRECTORY="${LOCAL_ROOT}"
export DATA_INPUT_DIR="${PH_SERVER_ROOT}/xli/reproducibility/out"
export SCRIPT_DIR="${PH_SERVER_ROOT}/xli/reproducibility/script"
export LOCAL_DATA_DIRECTORY="${DATA_INPUT_DIR}"
export DATA_OUTPUT_DIR="${PH_SERVER_ROOT}/fmriprep/Lei_working/Finalizing/Minimal/data"  # where to save output data on server
export WORKING_DIR="${PH_SERVER_ROOT}/fmriprep/Lei_working/Finalizing/Minimal/"  # where to store working files
export ROI_BASE="${PH_SERVER_ROOT}/fmriprep/Lei_working/testing/ICC_Scan_duration/All_sessions"
export RUN_ICC_OUTPUT="ICC_1000_All_pearson"
export CPAC_OUTPUT_DIR="${DATA_INPUT_DIR}/cpac_fmriprep_v2"  # C-PAC outputs
export FMRIPREP_OUTPUT_DIR="${PH_SERVER_ROOT}/fmriprep/Lei_working/FINAL_preprocessed_2021/fmriprep_default"  # fmriprep outputs
export GOOD_SUB_LIST_FILE="${PH_SERVER_ROOT}/fmriprep/Lei_working/FD_testing/HBN_testing/good_30.txt"  # list of subjects to use
```

with all of the necessary environment variables in the root of the project (I'd probably call it `.pipelineharmonizationrc`, and add something like

> ### Before running
> 1. Update `.pipelineharmonizationrc` with the applicable paths for your data.
> 2. Run `source .pipelineharmonizationrc` before running any scripts.
> 3. Run scripts (and IDEs like RStudio or Jupyter) from a terminal in which you've sourced `.pipelineharmonizationrc`

to the README.

Then someone using the scripts only has to update that one file instead of going into each file and updating the paths.

File: figure/3/extract_ROI.sh (hunk 1 above)
Comment (XinhuiLi) [reply figure/3/extract_ROI.sh]: Good catch!

**Comment by XinhuiLi** [comment]
Huge thanks Jon! I know I can always trust you! ❤️ Could you please review my changes one more time?

File: figure/2/plot_fig2.py [hunk 25]
```diff
@@ -452,19 +452,19 @@ def label(x, color, label):
     simpleplot=False
     atlases=['200','600','1000']
 
-    wd_path='/Users/xinhui.li/Documents/reproducibility/LA/Reproducibility_Analysis'
+    wd_path=f'{os.environ.get("BASEFOLDER")}'
```
Comment (shnizzedy) [inline figure/2/plot_fig2.py]: No need for an f-string here since there's nothing extra, could just do

```Python
wd_path=os.environ.get('BASEFOLDER')
```

**Review by shnizzedy — APPROVED** [review]
I finally finished going through the manuscript. I think it's the best paper I've ever been a coauthor on. Thanks for doing such nice work! :clap: :champagne: 

---

I made some changes that I would make to this repo and opened a PR (https://github.com/XinhuiLi/PipelineHarmonization/pull/2), but feel free to drop any you don't want.

If I have a chance, I'll try to simplify the structure of the tree / all those paths and actually run all these scripts with the environment variables. Is there any reason that some of these are local and some are on a server (other than that's where you / Lei ran them), or can we do everything on one machine?

**Comment by shnizzedy** [comment]
Oh, a couple other questions:

Are all these scripts intended to run on a server or local machine or in a C-PAC image or some of each or user's choice?

Also are there any files that are needed but not included, like maybe https://github.com/XinhuiLi/PipelineHarmonization/blob/893b53b2f7ffc0fc3243ce423252d9a13e696b33/figure/s1/abcd.py#L94-L97 these templates?

(and `/usr/local/fsl/data/standard/MNI152_T1_2mm_brain.nii.gz` on Lisa is probably the same as `/usr/share/data/fsl-mni152-templates/MNI152_T1_2mm_brain.nii.gz` in the C-PAC image, right?)

**Comment by XinhuiLi** [comment]
> Are all these scripts intended to run on a server or local machine or in a C-PAC image or some of each or user's choice?

Users can run on any machine. We usually do computation (like ICC) on server and download results from server to local machine to plot figures. 

> Also are there any files that are needed but not included, like maybe
> 
https://github.com/XinhuiLi/PipelineHarmonization/blob/893b53b2f7ffc0fc3243ce423252d9a13e696b33/figure/s1/abcd.py#L94-L97
> 
> these templates?

Good catch! I'll add notes for these templates.

> (and `/usr/local/fsl/data/standard/MNI152_T1_2mm_brain.nii.gz` on Lisa is probably the same as `/usr/share/data/fsl-mni152-templates/MNI152_T1_2mm_brain.nii.gz` in the C-PAC image, right?)

Yes!

**Pull request merged**